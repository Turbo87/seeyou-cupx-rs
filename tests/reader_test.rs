use insta::assert_compact_debug_snapshot;
use seeyou_cupx::CupxFile;
use std::io::Read;

#[test]
fn test_westalpen() {
    let (mut cupx, warnings) = CupxFile::from_path("tests/fixtures/westalpen_de.cupx").unwrap();
    assert_eq!(cupx.waypoints().len(), 126);
    assert_eq!(cupx.tasks().len(), 0);
    assert_eq!(warnings.len(), 0);

    let mut image_files = cupx.picture_names().collect::<Vec<_>>();
    image_files.sort();
    insta::assert_debug_snapshot!(image_files);

    let mut reader = cupx.read_picture("2_1034.jpg").unwrap();
    let mut buffer = Vec::new();
    reader.read_to_end(&mut buffer).unwrap();
    insta::assert_binary_snapshot!("2_1034.jpg", buffer);
}

#[test]
fn test_ec25_no_pictures_zip() {
    let (cupx, warnings) = CupxFile::from_path("tests/fixtures/EC25_no_pictures_zip.cupx").unwrap();
    assert_eq!(cupx.waypoints().len(), 221);
    assert_eq!(cupx.tasks().len(), 0);
    assert_compact_debug_snapshot!(warnings, @"[NoPicturesArchive]");
    assert_eq!(cupx.picture_names().count(), 0);
}
