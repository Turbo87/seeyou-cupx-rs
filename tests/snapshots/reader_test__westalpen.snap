---
source: tests/reader_test.rs
expression: image_files
---
[
    "1_101_osm.jpg",
    "1_135_osm.jpg",
    "1_20_osm.jpg",
    "1_21_osm.jpg",
    "1_22_osm.jpg",
    "1_23_osm.jpg",
    "1_24_osm.jpg",
    "1_25_osm.jpg",
    "1_26_osm.jpg",
    "1_27_osm.jpg",
    "1_28_osm.jpg",
    "1_29_osm.jpg",
    "1_30_osm.jpg",
    "1_32_osm.jpg",
    "1_33_osm.jpg",
    "1_34_osm.jpg",
    "1_35_osm.jpg",
    "1_36_osm.jpg",
    "1_38_osm.jpg",
    "1_39_osm.jpg",
    "1_40_osm.jpg",
    "1_41_osm.jpg",
    "1_42_osm.jpg",
    "1_43_osm.jpg",
    "1_44_osm.jpg",
    "1_4557_osm.jpg",
    "1_45_osm.jpg",
    "1_46_osm.jpg",
    "1_4727_osm.jpg",
    "1_4728_osm.jpg",
    "1_4730_osm.jpg",
    "1_4792_osm.jpg",
    "1_4798_osm.jpg",
    "1_47_osm.jpg",
    "1_4800_osm.jpg",
    "1_4802_osm.jpg",
    "1_4804_osm.jpg",
    "1_4805_osm.jpg",
    "1_4806_osm.jpg",
    "1_4809_osm.jpg",
    "1_4849_osm.jpg",
    "1_4867_osm.jpg",
    "1_486_osm.jpg",
    "1_4873_osm.jpg",
    "1_48_osm.jpg",
    "1_4906_osm.jpg",
    "1_4907_osm.jpg",
    "1_4957_osm.jpg",
    "1_4960_osm.jpg",
    "1_4961_osm.jpg",
    "1_4963_osm.jpg",
    "1_4964_osm.jpg",
    "1_4965_osm.jpg",
    "1_4966_osm.jpg",
    "1_4967_osm.jpg",
    "1_4968_osm.jpg",
    "1_4969_osm.jpg",
    "1_4970_osm.jpg",
    "1_4972_osm.jpg",
    "1_4974_osm.jpg",
    "1_4975_osm.jpg",
    "1_4977_osm.jpg",
    "1_4978_osm.jpg",
    "1_4979_osm.jpg",
    "1_4980_osm.jpg",
    "1_4981_osm.jpg",
    "1_4983_osm.jpg",
    "1_4984_osm.jpg",
    "1_4985_osm.jpg",
    "1_4986_osm.jpg",
    "1_4987_osm.jpg",
    "1_4988_osm.jpg",
    "1_49_osm.jpg",
    "1_500_osm.jpg",
    "1_50_osm.jpg",
    "1_51_osm.jpg",
    "1_5225_osm.jpg",
    "1_5261_osm.jpg",
    "1_5262_osm.jpg",
    "1_5283_osm.jpg",
    "1_5288_osm.jpg",
    "1_5289_osm.jpg",
    "1_5294_osm.jpg",
    "1_5295_osm.jpg",
    "1_52_osm.jpg",
    "1_5302_osm.jpg",
    "1_55_osm.jpg",
    "1_56_osm.jpg",
    "1_57_osm.jpg",
    "1_58_osm.jpg",
    "1_59_osm.jpg",
    "1_60_osm.jpg",
    "1_61_osm.jpg",
    "1_62_osm.jpg",
    "1_63_osm.jpg",
    "1_64_osm.jpg",
    "1_65_osm.jpg",
    "1_66_osm.jpg",
    "1_689_osm.jpg",
    "1_68_osm.jpg",
    "1_691_osm.jpg",
    "1_693_osm.jpg",
    "1_694_osm.jpg",
    "1_695_osm.jpg",
    "1_69_osm.jpg",
    "1_702_osm.jpg",
    "1_703_osm.jpg",
    "1_70_osm.jpg",
    "1_72_osm.jpg",
    "1_73_osm.jpg",
    "1_74_osm.jpg",
    "1_75_osm.jpg",
    "1_76_osm.jpg",
    "1_77_osm.jpg",
    "1_78_osm.jpg",
    "1_79_osm.jpg",
    "1_80_osm.jpg",
    "1_81_osm.jpg",
    "1_82_osm.jpg",
    "1_84_osm.jpg",
    "1_85_osm.jpg",
    "1_86_osm.jpg",
    "1_87_osm.jpg",
    "1_88_osm.jpg",
    "1_89_osm.jpg",
    "1_90_osm.jpg",
    "2_1034.jpg",
    "2_1127.jpg",
    "2_1129.jpg",
    "2_1990.jpg",
    "2_2021.jpg",
    "2_2058.jpg",
    "2_2120.jpg",
    "2_2409.jpg",
    "2_2411.jpg",
    "2_2416.jpg",
    "2_2435.jpg",
    "2_2436.jpg",
    "2_2452.jpg",
    "2_2591.jpg",
    "2_2606.jpg",
    "2_2607.jpg",
    "2_2612.jpg",
    "2_2637.jpg",
    "2_2645.jpg",
    "2_2657.jpg",
    "2_2860.jpg",
    "2_2874.jpg",
    "2_3107.jpg",
    "2_3126.jpg",
    "2_3156.jpg",
    "2_3348.jpg",
    "2_3386.jpg",
    "2_3398.jpg",
    "2_3481.jpg",
    "2_3485.jpg",
    "2_3542.jpg",
    "2_3577.jpg",
    "2_3578.jpg",
    "2_3966.jpg",
    "2_3970.jpg",
    "2_52.jpg",
    "2_54.jpg",
    "2_56.jpg",
    "2_58.jpg",
    "2_60.jpg",
    "2_62.jpg",
    "2_64.jpg",
    "2_74.jpg",
    "2_76.jpg",
    "2_85.jpg",
    "2_87.jpg",
]
